sources/Afup/Comptabilite/Facture.php

```diff
@@ -599,4 +602,18 @@ function envoyerFacture($reference)
 
         return $ok;
     }
+
+    public function encryptLink($data)
+    {
+        $iv = openssl_random_pseudo_bytes(openssl_cipher_iv_length($this->cipher));
+
+        return base64_encode(openssl_encrypt($data, $this->cipher, $this->key, 0, $iv) . '::' . $iv);
```

Comment:
Utilisation  de URLBase64 car la valeur est utilisée dans une URL.

```suggestion
return str_replace(['+','/'] , ['-','_'], base64_encode(openssl_encrypt($data, $this->cipher, $this->key, 0, $iv) . '::' . $iv));
```